bundles/org.openhab.voice.marytts/bnd.bnd

```diff
@@ -0,0 +1,15 @@
+Bundle-SymbolicName: ${project.artifactId}
+DynamicImport-Package: *
```

Comment:
I think this will do more harm than good.

```suggestion
```

Reply:
Removing leads to ClassNotFound exceptions during reload. To make it work again without this line, I would need to add 
`Import-Package: org.apache.log4j, org.slf4j, org.openhab.core.voice, org.openhab.core.audio, org.osgi.framework;version="[1.8,2)"`

I cannot judge what is better.

Reply:
The defaults we use are listed here: https://github.com/openhab/openhab-addons/blob/main/pom.xml#L145-L166
By adding this file you override all the defaults.

Reply:
The dynamic import only makes sense if you want to import each and every class exported in your application. That results in unwanted issues because it will also cause the bundle to be refreshed/reloaded whenever other bundles are installed.

Reply:
Ok, then the dynamic import should be avoided.
Do you think it is wrong to add the bnd.bnd at all? Should we inject the additional information via plugin configuration in pom? It feels wrong to write down the imports all over again, as there is another place where they are specified.

Reply:
Yes it's preferred to reuse the defaults as much as possible otherwise it will likely break in the future. We could also add a file that has the defaults which could then be [included](https://bnd.bndtools.org/instructions/include.html) by add-ons that use bnd.bnd files. We also use  an [include](https://github.com/openhab/openhab-core/blob/ab96a6ba5535b71b8929776b253c35db8cc91f15/itests/itest-common.bndrun#L1) in the core itests

Another way could be to add this information by adding some annotations. These will then also get added to the generated MANIFEST.MF file by Bnd. E.g. like in this [package-info.java](https://github.com/openhab/openhab-addons/blob/f87ff2f9cef16a7110403e23174bc2333950efba/bundles/org.openhab.automation.groovyscripting/src/main/java/org/openhab/automation/groovyscripting/internal/package-info.java#L13). The `org.osgi.annotation.bundle` package contains a few additional annotation interfaces that can be used with these annotations. Bnd also provides [SPI Annotations](https://bnd.bndtools.org/chapters/240-spi-annotations.html) (part of the `biz.aQute.bnd:biz.aQute.bnd.annotation` Maven dependency).